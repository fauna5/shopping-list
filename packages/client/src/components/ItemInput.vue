<template>
  <div>
    <input class="item-input" placeholder="add to list" v-model="itemTitle" v-on:keyup="name"/>
  </div>
</template>

<script>
import { mapState } from 'vuex';
import ListItem from './ListItem';
import { ADD } from '../store/actions';

export default {
  name: 'item-input',
  data() {
    return {
      itemTitle: '',
    }
  },
  methods: {
    name: function(e) {
      if(e.code === 'Enter'){
        this.$store.dispatch(ADD, this.itemTitle)
      }
    }
  },
};
</script>

<style scoped>
.item-input {
  width: calc(100% - 33px);
  background: #EEE;
  font-size: 18px;
  padding: 10px;
  outline: none;
  margin: 10px 5px 5px 5px;
  border: 1px solid #ddd
}
</style>