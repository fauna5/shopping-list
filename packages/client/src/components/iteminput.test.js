import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import createStore from '../store/store';

import ItemInput from './ItemInput';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ItemInput', () => {
  it('renders', () => {
    const wrapper = shallowMount(ItemInput);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should dispatch item add on click', () => {
    const add = jest.fn();

    const wrapper = shallowMount(ItemInput, {
      store: new Vuex.Store({
        actions: { ADD: add },
      }),
      localVue,
    });
    const input = wrapper.find('input');
    input.setValue('new');
    input.trigger('keyup', { code: 'Enter' });
    expect(add).toHaveBeenCalledWith(expect.anything(), 'new', undefined);
  });
});
