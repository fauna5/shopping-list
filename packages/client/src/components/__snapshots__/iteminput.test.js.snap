// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemInput renders 1`] = `
<div>
  <input
    class="item-input"
    placeholder="add to list"
  />
</div>
`;
